[C,test/socket.js,362]                             expect(statusCode).to.equal(404);
[C,test/socket.js,363]                             client.disconnect();
[C,test/socket.js,364]                             server.stop(done);
[C,test/socket.js,365]                         });
[C,test/socket.js,366]                     });
[C,test/socket.js,367]                 });
[C,test/socket.js,368]             });
[C,test/socket.js,369]         });
[C,test/socket.js,370] 
[C,test/socket.js,371]         it('errors on missing id', (done) => {
[C,test/socket.js,372] 
[C,test/socket.js,373]             const server = new Hapi.Server();
[C,test/socket.js,374]             server.connection();
[C,test/socket.js,375]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/socket.js,376] 
[C,test/socket.js,377]                 expect(err).to.not.exist();
[C,test/socket.js,378] 
[C,test/socket.js,379]                 server.route({
[C,test/socket.js,380]                     method: 'GET',
[C,test/socket.js,381]                     path: '/',
[C,test/socket.js,382]                     handler: function (request, reply) {
[C,test/socket.js,383] 
[C,test/socket.js,384]                         return reply('hello');
[C,test/socket.js,385]                     }
[C,test/socket.js,386]                 });
[C,test/socket.js,387] 
[C,test/socket.js,388]                 server.start((err) => {
[C,test/socket.js,389] 
[A,test/socket.js,390]                     expect(err).to.not.exist();
[C,test/socket.js,391]                     const client = new Ws('http://localhost:' + server.info.port);
[C,test/socket.js,392] 
[C,test/socket.js,393]                     client.on('message', (data, flags) => {
[C,test/socket.js,394] 
[C,test/socket.js,395]                         const message = JSON.parse(data);
[C,test/socket.js,396]                         expect(message.payload).to.deep.equal({
[C,test/socket.js,397]                             error: 'Bad Request',
[C,test/socket.js,398]                             message: 'Message missing id'
[C,test/socket.js,399]                         });
[C,test/socket.js,400] 
[C,test/socket.js,401]                         expect(message.statusCode).to.equal(400);
[C,test/socket.js,402]                         expect(message.type).to.equal('request');
[C,test/socket.js,403] 
[C,test/socket.js,404]                         client.close();
[C,test/socket.js,405]                         server.stop(done);
@@ -412,30 +423,31 @@ describe('Socket', () => {
[C,test/socket.js,423]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/socket.js,424] 
[C,test/socket.js,425]                 expect(err).to.not.exist();
[C,test/socket.js,426] 
[C,test/socket.js,427]                 server.route({
[C,test/socket.js,428]                     method: 'GET',
[C,test/socket.js,429]                     path: '/',
[C,test/socket.js,430]                     handler: function (request, reply) {
[C,test/socket.js,431] 
[C,test/socket.js,432]                         return reply('hello');
[C,test/socket.js,433]                     }
[C,test/socket.js,434]                 });
[C,test/socket.js,435] 
[C,test/socket.js,436]                 server.start((err) => {
[C,test/socket.js,437] 
[A,test/socket.js,438]                     expect(err).to.not.exist();
[C,test/socket.js,439]                     const client = new Ws('http://localhost:' + server.info.port);
[C,test/socket.js,440] 
[C,test/socket.js,441]                     client.on('message', (data, flags) => {
[C,test/socket.js,442] 
[C,test/socket.js,443]                         const message = JSON.parse(data);
[C,test/socket.js,444]                         expect(message.payload.message).to.equal('Connection is not initialized');
[C,test/socket.js,445] 
[C,test/socket.js,446]                         client.close();
[C,test/socket.js,447]                         server.stop(done);
[C,test/socket.js,448]                     });
[C,test/socket.js,449] 
[C,test/socket.js,450]                     client.on('open', () => {
[C,test/socket.js,451] 
[C,test/socket.js,452]                         client.send(JSON.stringify({ id: 1, type: 'request', path: '/' }), (err) => {
[C,test/socket.js,453] 
@@ -453,30 +465,31 @@ describe('Socket', () => {
[C,test/socket.js,465]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/socket.js,466] 
[C,test/socket.js,467]                 expect(err).to.not.exist();
[C,test/socket.js,468] 
[C,test/socket.js,469]                 server.route({
[C,test/socket.js,470]                     method: 'GET',
[C,test/socket.js,471]                     path: '/',
[C,test/socket.js,472]                     handler: function (request, reply) {
[C,test/socket.js,473] 
[C,test/socket.js,474]                         return reply('hello');
[C,test/socket.js,475]                     }
[C,test/socket.js,476]                 });
[C,test/socket.js,477] 
[C,test/socket.js,478]                 server.start((err) => {
[C,test/socket.js,479] 
[A,test/socket.js,480]                     expect(err).to.not.exist();
[C,test/socket.js,481]                     const client = new Ws('http://localhost:' + server.info.port);
[C,test/socket.js,482] 
[C,test/socket.js,483]                     client.on('message', (data, flags) => {
[C,test/socket.js,484] 
[C,test/socket.js,485]                         const message = JSON.parse(data);
[C,test/socket.js,486]                         if (message.id !== 2) {
[C,test/socket.js,487]                             return;
[C,test/socket.js,488]                         }
[C,test/socket.js,489] 
[C,test/socket.js,490]                         expect(message.payload).to.deep.equal({
[C,test/socket.js,491]                             error: 'Bad Request',
[C,test/socket.js,492]                             message: 'Message missing method'
[C,test/socket.js,493]                         });
[C,test/socket.js,494] 
[C,test/socket.js,495]                         expect(message.statusCode).to.equal(400);
@@ -508,30 +521,31 @@ describe('Socket', () => {
[C,test/socket.js,521]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/socket.js,522] 
[C,test/socket.js,523]                 expect(err).to.not.exist();
[C,test/socket.js,524] 
[C,test/socket.js,525]                 server.route({
[C,test/socket.js,526]                     method: 'GET',
[C,test/socket.js,527]                     path: '/',
[C,test/socket.js,528]                     handler: function (request, reply) {
[C,test/socket.js,529] 
[C,test/socket.js,530]                         return reply('hello');
[C,test/socket.js,531]                     }
[C,test/socket.js,532]                 });
[C,test/socket.js,533] 
[C,test/socket.js,534]                 server.start((err) => {
[C,test/socket.js,535] 
[A,test/socket.js,536]                     expect(err).to.not.exist();
[C,test/socket.js,537]                     const client = new Ws('http://localhost:' + server.info.port);
[C,test/socket.js,538] 
[C,test/socket.js,539]                     client.on('message', (data, flags) => {
[C,test/socket.js,540] 
[C,test/socket.js,541]                         const message = JSON.parse(data);
[C,test/socket.js,542]                         if (message.id !== 2) {
[C,test/socket.js,543]                             return;
[C,test/socket.js,544]                         }
[C,test/socket.js,545] 
[C,test/socket.js,546]                         expect(message.payload).to.deep.equal({
[C,test/socket.js,547]                             error: 'Bad Request',
[C,test/socket.js,548]                             message: 'Message missing path'
[C,test/socket.js,549]                         });
[C,test/socket.js,550] 
[C,test/socket.js,551]                         expect(message.statusCode).to.equal(400);
@@ -563,30 +577,31 @@ describe('Socket', () => {
[C,test/socket.js,577]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/socket.js,578] 
[C,test/socket.js,579]                 expect(err).to.not.exist();
[C,test/socket.js,580] 
[C,test/socket.js,581]                 server.route({
[C,test/socket.js,582]                     method: 'GET',
[C,test/socket.js,583]                     path: '/',
[C,test/socket.js,584]                     handler: function (request, reply) {
[C,test/socket.js,585] 
[C,test/socket.js,586]                         return reply('hello');
[C,test/socket.js,587]                     }
[C,test/socket.js,588]                 });
[C,test/socket.js,589] 
[C,test/socket.js,590]                 server.start((err) => {
[C,test/socket.js,591] 
[A,test/socket.js,592]                     expect(err).to.not.exist();
[C,test/socket.js,593]                     const client = new Ws('http://localhost:' + server.info.port);
[C,test/socket.js,594] 
[C,test/socket.js,595]                     client.on('message', (data, flags) => {
[C,test/socket.js,596] 
[C,test/socket.js,597]                         const message = JSON.parse(data);
[C,test/socket.js,598]                         if (message.id !== 2) {
[C,test/socket.js,599]                             return;
[C,test/socket.js,600]                         }
[C,test/socket.js,601] 
[C,test/socket.js,602]                         expect(message.payload).to.deep.equal({
[C,test/socket.js,603]                             error: 'Bad Request',
[C,test/socket.js,604]                             message: 'Unknown message type'
[C,test/socket.js,605]                         });
[C,test/socket.js,606] 
[C,test/socket.js,607]                         expect(message.statusCode).to.equal(400);
@@ -609,30 +624,31 @@ describe('Socket', () => {
[C,test/socket.js,624]                     });
[C,test/socket.js,625]                 });
[C,test/socket.js,626]             });
[C,test/socket.js,627]         });
[C,test/socket.js,628] 
[C,test/socket.js,629]         it('errors on incorrect version', (done) => {
[C,test/socket.js,630] 
[C,test/socket.js,631]             const server = new Hapi.Server();
[C,test/socket.js,632]             server.connection();
[C,test/socket.js,633]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/socket.js,634] 
[C,test/socket.js,635]                 expect(err).to.not.exist();
[C,test/socket.js,636] 
[C,test/socket.js,637]                 server.start((err) => {
[C,test/socket.js,638] 
[A,test/socket.js,639]                     expect(err).to.not.exist();
[C,test/socket.js,640]                     const client = new Ws('http://localhost:' + server.info.port);
[C,test/socket.js,641] 
[C,test/socket.js,642]                     client.on('message', (data, flags) => {
[C,test/socket.js,643] 
[C,test/socket.js,644]                         const message = JSON.parse(data);
[C,test/socket.js,645]                         expect(message.payload).to.deep.equal({
[C,test/socket.js,646]                             error: 'Bad Request',
[C,test/socket.js,647]                             message: 'Incorrect protocol version (expected 2 but received 1)'
[C,test/socket.js,648]                         });
[C,test/socket.js,649] 
[C,test/socket.js,650]                         expect(message.statusCode).to.equal(400);
[C,test/socket.js,651] 
[C,test/socket.js,652]                         client.close();
[C,test/socket.js,653]                         server.stop(done);
[C,test/socket.js,654]                     });
@@ -646,30 +662,31 @@ describe('Socket', () => {
[C,test/socket.js,662]                     });
[C,test/socket.js,663]                 });
[C,test/socket.js,664]             });
[C,test/socket.js,665]         });
[C,test/socket.js,666] 
[C,test/socket.js,667]         it('errors on missing version', (done) => {
[C,test/socket.js,668] 
[C,test/socket.js,669]             const server = new Hapi.Server();
[C,test/socket.js,670]             server.connection();
[C,test/socket.js,671]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/socket.js,672] 
[C,test/socket.js,673]                 expect(err).to.not.exist();
[C,test/socket.js,674] 
[C,test/socket.js,675]                 server.start((err) => {
[C,test/socket.js,676] 
[A,test/socket.js,677]                     expect(err).to.not.exist();
[C,test/socket.js,678]                     const client = new Ws('http://localhost:' + server.info.port);
[C,test/socket.js,679] 
[C,test/socket.js,680]                     client.on('message', (data, flags) => {
[C,test/socket.js,681] 
[C,test/socket.js,682]                         const message = JSON.parse(data);
[C,test/socket.js,683]                         expect(message.payload).to.deep.equal({
[C,test/socket.js,684]                             error: 'Bad Request',
[C,test/socket.js,685]                             message: 'Incorrect protocol version (expected 2 but received none)'
[C,test/socket.js,686]                         });
[C,test/socket.js,687] 
[C,test/socket.js,688]                         expect(message.statusCode).to.equal(400);
[C,test/socket.js,689] 
[C,test/socket.js,690]                         client.close();
[C,test/socket.js,691]                         server.stop(done);
[C,test/socket.js,692]                     });
@@ -691,42 +708,45 @@ describe('Socket', () => {
[C,test/socket.js,708]             server.connection();
[C,test/socket.js,709] 
[C,test/socket.js,710]             const onMessage = function (socket, message, next) {
[C,test/socket.js,711] 
[C,test/socket.js,712]                 return next('b');
[C,test/socket.js,713]             };
[C,test/socket.js,714] 
[C,test/socket.js,715]             server.register({ register: Nes, options: { auth: false, onMessage: onMessage } }, (err) => {
[C,test/socket.js,716] 
[C,test/socket.js,717]                 expect(err).to.not.exist();
[C,test/socket.js,718] 
[C,test/socket.js,719]                 server.subscription('/{id}', {});
[C,test/socket.js,720] 
[C,test/socket.js,721]                 server.start((err) => {
[C,test/socket.js,722] 
[A,test/socket.js,723]                     expect(err).to.not.exist();
[C,test/socket.js,724]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/socket.js,725]                     client.connect(() => {
[C,test/socket.js,726] 
[C,test/socket.js,727]                         client.subscribe('/5', Hoek.ignore, (err) => {
[C,test/socket.js,728] 
[A,test/socket.js,729]                             expect(err).to.not.exist();
[C,test/socket.js,730]                             const handler = (update) => {
[C,test/socket.js,731] 
[C,test/socket.js,732]                                 client.unsubscribe('/5');
[C,test/socket.js,733]                                 client.unsubscribe('/6');
[C,test/socket.js,734] 
[C,test/socket.js,735]                                 client.message('a', (err, message) => {
[C,test/socket.js,736] 
[A,test/socket.js,737]                                     expect(err).to.not.exist();
[C,test/socket.js,738]                                     const listener = server.connections[0].plugins.nes._listener;
[C,test/socket.js,739]                                     const match = listener._router.route('sub', '/5');
[C,test/socket.js,740]                                     expect(match.route.subscribers._items).to.deep.equal({});
[C,test/socket.js,741] 
[C,test/socket.js,742]                                     client.disconnect();
[C,test/socket.js,743]                                     server.stop(done);
[C,test/socket.js,744]                                 });
[C,test/socket.js,745]                             };
[C,test/socket.js,746] 
[C,test/socket.js,747]                             client.subscribe('/6', handler, (err) => {
[C,test/socket.js,748] 
[C,test/socket.js,749]                                 expect(err).to.not.exist();
[C,test/socket.js,750]                                 server.publish('/6', 'b');
[C,test/socket.js,751]                             });
[C,test/socket.js,752]                         });
@@ -741,40 +761,42 @@ describe('Socket', () => {
[C,test/socket.js,761]             server.connection();
[C,test/socket.js,762] 
[C,test/socket.js,763]             const onMessage = function (socket, message, next) {
[C,test/socket.js,764] 
[C,test/socket.js,765]                 return next('b');
[C,test/socket.js,766]             };
[C,test/socket.js,767] 
[C,test/socket.js,768]             server.register({ register: Nes, options: { auth: false, onMessage: onMessage } }, (err) => {
[C,test/socket.js,769] 
[C,test/socket.js,770]                 expect(err).to.not.exist();
[C,test/socket.js,771] 
[C,test/socket.js,772]                 server.subscription('/{id}', {});
[C,test/socket.js,773] 
[C,test/socket.js,774]                 server.start((err) => {
[C,test/socket.js,775] 
[A,test/socket.js,776]                     expect(err).to.not.exist();
[C,test/socket.js,777]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/socket.js,778]                     client.connect(() => {
[C,test/socket.js,779] 
[C,test/socket.js,780]                         const handler = (update) => {
[C,test/socket.js,781] 
[C,test/socket.js,782]                             client.unsubscribe('/6');
[C,test/socket.js,783]                             client._send({ type: 'unsub', path: '/6' });
[C,test/socket.js,784] 
[C,test/socket.js,785]                             client.message('a', (err, message) => {
[C,test/socket.js,786] 
[A,test/socket.js,787]                                 expect(err).to.not.exist();
[C,test/socket.js,788]                                 const listener = server.connections[0].plugins.nes._listener;
[C,test/socket.js,789]                                 const match = listener._router.route('sub', '/6');
[C,test/socket.js,790]                                 expect(match.route.subscribers._items).to.deep.equal({});
[C,test/socket.js,791] 
[C,test/socket.js,792]                                 client.disconnect();
[C,test/socket.js,793]                                 server.stop(done);
[C,test/socket.js,794]                             });
[C,test/socket.js,795]                         };
[C,test/socket.js,796] 
[C,test/socket.js,797]                         client.subscribe('/6', handler, (err) => {
[C,test/socket.js,798] 
[C,test/socket.js,799]                             expect(err).to.not.exist();
[C,test/socket.js,800]                             server.publish('/6', 'b');
[C,test/socket.js,801]                         });
[C,test/socket.js,802]                     });
@@ -792,30 +814,31 @@ describe('Socket', () => {
[C,test/socket.js,814]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/socket.js,815] 
[C,test/socket.js,816]                 expect(err).to.not.exist();
[C,test/socket.js,817] 
[C,test/socket.js,818]                 server.route({
[C,test/socket.js,819]                     method: 'GET',
[C,test/socket.js,820]                     path: '/',
[C,test/socket.js,821]                     handler: function (request, reply) {
[C,test/socket.js,822] 
[C,test/socket.js,823]                         return reply(request.socket.id);
[C,test/socket.js,824]                     }
[C,test/socket.js,825]                 });
[C,test/socket.js,826] 
[C,test/socket.js,827]                 server.start((err) => {
[C,test/socket.js,828] 
[A,test/socket.js,829]                     expect(err).to.not.exist();
[C,test/socket.js,830]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/socket.js,831]                     client.connect(() => {
[C,test/socket.js,832] 
[C,test/socket.js,833]                         client.request('/', (err, payload, statusCode, headers) => {
[C,test/socket.js,834] 
[C,test/socket.js,835]                             expect(err).to.not.exist();
[C,test/socket.js,836]                             expect(payload).to.equal(client.id);
[C,test/socket.js,837] 
[C,test/socket.js,838]                             client.disconnect();
[C,test/socket.js,839]                             server.stop(done);
[C,test/socket.js,840]                         });
[C,test/socket.js,841]                     });
[C,test/socket.js,842]                 });
[C,test/socket.js,843]             });
[C,test/socket.js,844]         });
@@ -827,30 +850,31 @@ describe('Socket', () => {
[C,test/socket.js,850]             server.register({ register: Nes, options: { auth: false, headers: '*' } }, (err) => {
[C,test/socket.js,851] 
[C,test/socket.js,852]                 expect(err).to.not.exist();
[C,test/socket.js,853] 
[C,test/socket.js,854]                 server.route({
[C,test/socket.js,855]                     method: 'GET',
[C,test/socket.js,856]                     path: '/',
[C,test/socket.js,857]                     handler: function (request, reply) {
[C,test/socket.js,858] 
[C,test/socket.js,859]                         return reply('hello ' + request.headers.a);
[C,test/socket.js,860]                     }
[C,test/socket.js,861]                 });
[C,test/socket.js,862] 
[C,test/socket.js,863]                 server.start((err) => {
[C,test/socket.js,864] 
[A,test/socket.js,865]                     expect(err).to.not.exist();
[C,test/socket.js,866]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/socket.js,867]                     client.connect(() => {
[C,test/socket.js,868] 
[C,test/socket.js,869]                         client.request({ path: '/', headers: { a: 'b' } }, (err, payload, statusCode, headers) => {
[C,test/socket.js,870] 
[C,test/socket.js,871]                             expect(err).to.not.exist();
[C,test/socket.js,872]                             expect(payload).to.equal('hello b');
[C,test/socket.js,873]                             expect(statusCode).to.equal(200);
[C,test/socket.js,874]                             expect(headers).to.contain({ 'content-type': 'text/html; charset=utf-8' });
[C,test/socket.js,875] 
[C,test/socket.js,876]                             client.disconnect();
[C,test/socket.js,877]                             server.stop(done);
[C,test/socket.js,878]                         });
[C,test/socket.js,879]                     });
[C,test/socket.js,880]                 });
@@ -864,30 +888,31 @@ describe('Socket', () => {
[C,test/socket.js,888]             server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/socket.js,889] 
[C,test/socket.js,890]                 expect(err).to.not.exist();
[C,test/socket.js,891] 
[C,test/socket.js,892]                 server.route({
[C,test/socket.js,893]                     method: 'GET',
[C,test/socket.js,894]                     path: '/',
[C,test/socket.js,895]                     handler: function (request, reply) {
[C,test/socket.js,896] 
[C,test/socket.js,897]                         return reply('hello');
[C,test/socket.js,898]                     }
[C,test/socket.js,899]                 });
[C,test/socket.js,900] 
[C,test/socket.js,901]                 server.start((err) => {
[C,test/socket.js,902] 
[A,test/socket.js,903]                     expect(err).to.not.exist();
[C,test/socket.js,904]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/socket.js,905]                     client.connect(() => {
[C,test/socket.js,906] 
[C,test/socket.js,907]                         client.request({ path: '/', headers: { Authorization: 'something' } }, (err, payload, statusCode, headers) => {
[C,test/socket.js,908] 
[C,test/socket.js,909]                             expect(err).to.exist();
[C,test/socket.js,910]                             expect(err.message).to.equal('Cannot include an Authorization header');
[C,test/socket.js,911] 
[C,test/socket.js,912]                             client.disconnect();
[C,test/socket.js,913]                             server.stop(done);
[C,test/socket.js,914]                         });
[C,test/socket.js,915]                     });
[C,test/socket.js,916]                 });
[C,test/socket.js,917]             });
[C,test/socket.js,918]         });
@@ -899,30 +924,31 @@ describe('Socket', () => {
[C,test/socket.js,924] 
[C,test/socket.js,925]             const onMessage = function (socket, message, reply) {
[C,test/socket.js,926] 
[C,test/socket.js,927]                 expect(message).to.equal('winning');
[C,test/socket.js,928]                 reply('hello');
[C,test/socket.js,929]             };
[C,test/socket.js,930] 
[C,test/socket.js,931]             const server = new Hapi.Server();
[C,test/socket.js,932]             server.connection();
[C,test/socket.js,933]             server.register({ register: Nes, options: { onMessage: onMessage } }, (err) => {
[C,test/socket.js,934] 
[C,test/socket.js,935]                 expect(err).to.not.exist();
[C,test/socket.js,936] 
[C,test/socket.js,937]                 server.start((err) => {
[C,test/socket.js,938] 
[A,test/socket.js,939]                     expect(err).to.not.exist();
[C,test/socket.js,940]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/socket.js,941]                     client.connect(() => {
[C,test/socket.js,942] 
[C,test/socket.js,943]                         client.message('winning', (err, response) => {
[C,test/socket.js,944] 
[C,test/socket.js,945]                             expect(err).to.not.exist();
[C,test/socket.js,946]                             expect(response).to.equal('hello');
[C,test/socket.js,947]                             client.disconnect();
[C,test/socket.js,948]                             server.stop(done);
[C,test/socket.js,949]                         });
[C,test/socket.js,950]                     });
[C,test/socket.js,951]                 });
[C,test/socket.js,952]             });
[C,test/socket.js,953]         });
[C,test/socket.js,954] 
@@ -932,30 +958,31 @@ describe('Socket', () => {
[C,test/socket.js,958] 
[C,test/socket.js,959]             const onMessage = function (socket, message, reply) {
[C,test/socket.js,960] 
[C,test/socket.js,961]                 expect(message).to.equal('winning');
[C,test/socket.js,962]                 reply(new Error('failed'));
[C,test/socket.js,963]             };
[C,test/socket.js,964] 
[C,test/socket.js,965]             const server = new Hapi.Server();
[C,test/socket.js,966]             server.connection();
[C,test/socket.js,967]             server.register({ register: Nes, options: { onMessage: onMessage } }, (err) => {
[C,test/socket.js,968] 
[C,test/socket.js,969]                 expect(err).to.not.exist();
[C,test/socket.js,970] 
[C,test/socket.js,971]                 server.start((err) => {
[C,test/socket.js,972] 
[A,test/socket.js,973]                     expect(err).to.not.exist();
[C,test/socket.js,974]                     client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/socket.js,975]                     client.connect(() => {
[C,test/socket.js,976] 
[C,test/socket.js,977]                         client.message('winning', (err, response) => {
[C,test/socket.js,978] 
[C,test/socket.js,979]                             expect(err).to.exist();
[C,test/socket.js,980]                             expect(err.message).to.equal('An internal server error occurred');
[C,test/socket.js,981]                             expect(err.statusCode).to.equal(500);
[C,test/socket.js,982]                             expect(response).to.not.exist();
[C,test/socket.js,983]                             client.disconnect();
[C,test/socket.js,984]                             server.stop(done);
[C,test/socket.js,985]                         });
[C,test/socket.js,986]                     });
[C,test/socket.js,987]                 });
[C,test/socket.js,988]             });
@@ -967,57 +994,59 @@ describe('Socket', () => {
[C,test/socket.js,994] 
[C,test/socket.js,995]             const onMessage = function (socket, message, reply) {
[C,test/socket.js,996] 
[C,test/socket.js,997]                 expect(message).to.equal('winning');
[C,test/socket.js,998]                 reply(Boom.badRequest('failed'));
[C,test/socket.js,999]             };
[C,test/socket.js,1000] 
[C,test/socket.js,1001]             const server = new Hapi.Server();
[C,test/socket.js,1002]             server.connection();
[C,test/socket.js,1003]             server.register({ register: Nes, options: { onMessage: onMessage } }, (err) => {
[C,test/socket.js,1004] 
[C,test/socket.js,1005]                 expect(err).to.not.exist();
[C,test/socket.js,1006] 
[C,test/socket.js,1007]                 server.start((err) => {
[C,test/socket.js,1008] 
[A,test/socket.js,1009]                     expect(err).to.not.exist();
[C,test/socket.js,1010]                     client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/socket.js,1011]                     client.connect(() => {
[C,test/socket.js,1012] 
[C,test/socket.js,1013]                         client.message('winning', (err, response) => {
[C,test/socket.js,1014] 
[C,test/socket.js,1015]                             expect(err).to.exist();
[C,test/socket.js,1016]                             expect(err.message).to.equal('failed');
[C,test/socket.js,1017]                             expect(err.statusCode).to.equal(400);
[C,test/socket.js,1018]                             expect(response).to.not.exist();
[C,test/socket.js,1019]                             client.disconnect();
[C,test/socket.js,1020]                             server.stop(done);
[C,test/socket.js,1021]                         });
[C,test/socket.js,1022]                     });
[C,test/socket.js,1023]                 });
[C,test/socket.js,1024]             });
[C,test/socket.js,1025]         });
[C,test/socket.js,1026] 
[C,test/socket.js,1027]         it('errors if missing onMessage callback', (done) => {
[C,test/socket.js,1028] 
[C,test/socket.js,1029]             const server = new Hapi.Server();
[C,test/socket.js,1030]             server.connection();
[C,test/socket.js,1031]             server.register({ register: Nes, options: {} }, (err) => {
[C,test/socket.js,1032] 
[C,test/socket.js,1033]                 expect(err).to.not.exist();
[C,test/socket.js,1034] 
[C,test/socket.js,1035]                 server.start((err) => {
[C,test/socket.js,1036] 
[A,test/socket.js,1037]                     expect(err).to.not.exist();
[C,test/socket.js,1038]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/socket.js,1039]                     client.connect(() => {
[C,test/socket.js,1040] 
[C,test/socket.js,1041]                         client.message('winning', (err, response) => {
[C,test/socket.js,1042] 
[C,test/socket.js,1043]                             expect(err).to.exist();
[C,test/socket.js,1044]                             expect(err.message).to.equal('Not Implemented');
[C,test/socket.js,1045]                             expect(err.statusCode).to.equal(501);
[C,test/socket.js,1046] 
[C,test/socket.js,1047]                             client.disconnect();
[C,test/socket.js,1048]                             server.stop(done);
[C,test/socket.js,1049]                         });
[C,test/socket.js,1050]                     });
[C,test/socket.js,1051]                 });
[C,test/socket.js,1052]             });